crates/sui-json-rpc/src/read_api.rs

```diff
@@ -580,6 +580,69 @@ impl ReadApiServer for ReadApi {
         })
     }
 
+    #[instrument(skip(self))]
+    async fn multi_get_object_info_for_tx_building(
+        &self,
+        object_ids: Vec<ObjectID>,
+    ) -> RpcResult<Vec<SuiTxObjectInfo>> {
+        // 2048 is the maximum number of input objects allowed in a transaction.
+        // Setting it as the limit so that one can always use one query to get all the latest versions
+        // of a given transaction.
+        const MAX_OBJECT_IDS: usize = 2048;
+        with_tracing!(async move {
+            if object_ids.len() > MAX_OBJECT_IDS {
+                return Err(SuiRpcInputError::SizeLimitExceeded(MAX_OBJECT_IDS.to_string()).into());
+            }
+            let state = self.state.clone();
```

Comment:
nit: is the clone necessary anymore? Seems like an artifact of issuing the `get_object` reads in parallel.